sdk/webpubsub/Azure.Messaging.WebPubSub.Client/src/EventSource/WebPubSubClientEventSource.cs

```diff
@@ -0,0 +1,155 @@
+        private const int ClientStartingId = 1;
+        private const int ClientStateChangedId = 2;
+        private const int WebSocketConnectingId = 3;
+        private const int WebSocketClosedId = 4;
+        private const int FailedToProcessMessageId = 5;
+        private const int FailedToReceiveBytesId = 6;
+        private const int FailedToChangeClientStateId = 7;
+        private const int StopRecoveryId = 8;
+        private const int FailedToRecoverConnectionId = 9;
+        private const int FailedToInvokeEventId = 10;
+        private const int ConnectionConnectedId = 11;
+        private const int ConnectionDisconnectedId = 12;
+        private const int FailedToReconnectId = 13;
+
+        public static WebPubSubClientEventSource Log { get; } = new WebPubSubClientEventSource();
+
+        [Event(1, Level = EventLevel.Verbose, Message = "Client starting.")]
+        public virtual void ClientStarting()
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(ClientStartingId);
+            }
+        }
+
+        [Event(2, Level = EventLevel.Verbose, Message = "The client state transient from the '{0}' state to the '{1}' state.")]
+        public virtual void ClientStateChanged(string newState, string currentState)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(ClientStateChangedId, currentState, newState);
+            }
+        }
+
+        [Event(3, Level = EventLevel.Verbose, Message = "A new WebSocket connection start to connect with subprotocol {0}.")]
+        public virtual void WebSocketConnecting(string subprotocol)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(WebSocketConnectingId, subprotocol);
+            }
+        }
+
+        [Event(4, Level = EventLevel.Verbose, Message = "WebSocket connection closed.")]
+        public virtual void WebSocketClosed()
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(WebSocketClosedId);
+            }
+        }
+
+        [Event(5, Level = EventLevel.Warning, Message = "An exception occurred while process message from the service. Error Message: {0}")]
+        public virtual void FailedToProcessMessage(string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(FailedToProcessMessageId, errorMessage);
+            }
+        }
+
+        [Event(6, Level = EventLevel.Informational, Message = "An exception occurred while receiving bytes. Error Message: {0}")]
+        public virtual void FailedToReceiveBytes(string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(FailedToReceiveBytesId, errorMessage);
+            }
+        }
+
+        [Event(7, Level = EventLevel.Warning, Message = "The client failed to transition from the '{0}' state to the '{1}' state because it was actually in the '{2}' state.")]
```

Comment:
```suggestion
        [Event(7, Level = EventLevel.Warning, Message = "The client failed to change from the '{0}' state to the '{1}' state because it was actually in the '{2}' state.")]
```